packages/openapi-v3/test/integration/operation-spec.test.ts

```diff
@@ -0,0 +1,58 @@
+import {expect} from '@loopback/testlab';
+import {model, property} from '@loopback/repository';
+import {param, requestBody, getControllerSpec, post} from '../../';
+
+describe('operation arguments', () => {
+  it('generate parameters and requestBody for operation', () => {
+    @model()
+    class User {
+      @property() name: string;
+      @property() password: number;
+    }
+
+    class MyController {
+      @post('/users/{location}')
+      async createUser(
+        @param.query.string('type') type: string,
```

Comment:
As a side note, do we every use the string given as an argument in any of our codebase anymore? Is it possible for us to get rid of it?

Reply:
@shimks sorry could you elaborate more about this? not sure I understand...:speak_no_evil:

Reply:
Eh, I think I'm overloading the scope of this task, ignore my last comment :p

Reply:
@shimks Did you mean to say `@param.query('type')` is good enough as we can infer `string` from the TS type?